taichi/struct/struct_metal.cpp

```diff
@@ -56,12 +106,17 @@ class StructCompiler {
           stride_str += " + " + ch_node_name + "::stride";
         }
         emit("  }}");
+        emit("");
       }
+      emit("  device byte* addr() {{ return addr_; }}");
+      emit("");
       if (stride_str.empty()) {
         // Is it possible for this to have no children?
```

Comment:
Good point - we should probably forbid non-place SNodes with no children in the struct compiler #636